### PR TITLE
Updated to latest version of PTB and created workflow files

### LINKED ISSUE #8: NC: Update PTB and publish releases to pypi
See
* https://github.com/exasol/bucketfs-python/blob/main/.github/workflows/ci-cd.yml
* https://pypi.org/project/exasol-toolbox/

```[tasklist]
### Tasks
- [x] Update exasol python toolbox version to 0.5.0
- [ ] Fix build errors
- [ ] Publish releases to pypi
```

### PR DESCRIPTION
**Pull request by ckunki**
Closes  #8

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
+        run: poetry run nox -s lint
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:58]:
```suggestion
      - name: Run Linter
        run: poetry run nox -s lint
```

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:84]:
```suggestion
      - name: Run Type Check
```

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:106]:
```suggestion
      - name: Calculate Test Coverage
```

File: exasol/secret_store.py [hunk 4]
```diff
@@ -28,12 +28,16 @@ def close(self) -> None:
             self._con.close()
             self._con = None
 
-    def connection(self) -> sqlcipher.Connection:
+    def connection(
+        self,
+    ) -> sqlcipher.Connection:  # pylint: disable=E1101
```
Comment (kaklakariada) [inline exasol/secret_store.py:33]: Add a short explanation & justification for disabling pylint here

File: pyproject.toml [hunk 5]
```diff
@@ -53,21 +54,21 @@ force_grid_wrap = 2
...
-]
+    "exasol.toolbox.nox.tasks",
+    "test.conftest.*",
+    "test.*",
+    ]
 ignore_errors = true
+ignore_missing_imports = true
+
+[tool.pylint.format]
+max-line-length = 88
```
Comment (kaklakariada) [inline pyproject.toml:70]: This seems a bit short. 120 would also be OK in my opinion ;)

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: exasol/secret_store.py (hunk 4 above)
Comment (ckunki) [reply exasol/secret_store.py:33]: We were struggling with adding this explanation, but formatter inserted line breaks and moved disblable instruction then did not disable the linter anymore. :frowning: 
I add comment above. :slightly_smiling_face:

File: pyproject.toml (hunk 5 above)
Comment (ckunki) [reply pyproject.toml:70]: Ok, will update in next push.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (ckunki) [reply .github/workflows/checks.yml:58]: This file is added by python tool box. 
Though due to unexpeded build and compatibility problems I temporarily needed to apply some changes to the build files. I still propose to  create a ticket for PTB, though.

File: .github/workflows/checks.yml (hunk 2 above)
Comment (ckunki) [reply .github/workflows/checks.yml:84]: See comment above. I propose to create a ticket in github.com/exasol/python-toolbox/

File: .github/workflows/ci-cd.yml [hunk 6]
```diff
@@ -17,7 +17,7 @@ jobs:
     uses: ./.github/workflows/checks.yml
 
   cd-job:
-    name: Continues Delivery
+    name: Continuouss Delivery
```
Comment (kaklakariada) [inline .github/workflows/ci-cd.yml:20]:
```suggestion
    name: Continuous Delivery
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -3,33 +3,22 @@ name: CI
 on:
   push:
     branches-ignore:
+      - "github-pages/*"
+      - "gh-pages/*"
```
Comment (Nicoretti) [inline .github/workflows/ci.yml:7]: good point, could you mention that in a ticket for the toolbox?

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -3,33 +3,22 @@ name: CI
...
       - "main"
+      - "master"
```
Comment (Nicoretti) [inline .github/workflows/ci.yml:9]: same for this one :+1:

File: pyproject.toml [hunk 9]
```diff
@@ -53,21 +54,21 @@ force_grid_wrap = 2
 
 
 [tool.pylint.master]
-fail-under = 7.5
+fail-under = 6.2
```
Comment (Nicoretti) [inline pyproject.toml:57]: I think this one can be increased. I recommend setting it to the currently determined score, so deterioration can be avoided.

**Review by kaklakariada — APPROVED** [review]

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**